File: saucedemo-serenity/src/main/java/com/sauce/userinterfaces/LoginPage.java
package com.sauce.userinterfaces;

import net.serenitybdd.screenplay.targets.Target;
import org.openqa.selenium.By;

public class LoginPage {

	public static final Target USER = Target.the("Username").located(By.id("user-name"));
	public static final Target PASSWORD = Target.the("Enter you parssword").located(By.id("password"));
	public static final Target LOGIN_BUTTON = Target.the("Login button").located(By.id("login-button"));
	}

File: saucedemo-playwright/user-interfaces/login.page.ts
import {Page, Locator} from '@playwright/test'

class LoginPage {
    readonly page: Page
    user: Locator
    pass: Locator
    login: Locator

    constructor(page: Page) {
      if (!page) {
        throw new Error('The "page" object is undefined.');
      }
        this.page = page
        this.user = page.locator('input[id=\'user-name\']');
        this.pass = page.locator('input[id=\'password\']');
        this.login= page.locator('input[id=\'login-button\']');
        
    }

    public username() {
        return this.user;
      }

      public password() {
        return this.pass;
      }

      public loginBtn() {
        return this.login;
      }
}

export default LoginPage